src/decompiler.cc

```diff
@@ -16,54 +16,40 @@
 
 #include "src/decompiler.h"
 
-#include <algorithm>
-#include <array>
-#include <cassert>
-#include <cinttypes>
-#include <cstdarg>
-#include <cstdio>
-#include <iterator>
-#include <map>
-#include <numeric>
-#include <set>
-#include <string>
-#include <vector>
-#include <sstream>
-
-#include "src/cast.h"
-#include "src/common.h"
-#include "src/expr-visitor.h"
-#include "src/ir.h"
-#include "src/ir-util.h"
-#include "src/literal.h"
-#include "src/generate-names.h"
-#include "src/stream.h"
+#include "src/decompiler-ast.inl"
```

Comment:
Why `.inl`? Just to move the AST out-of-line but still build it as an object file?

Reply:
Because this code is entirely private to the decompiler, putting it in an object file means pulling out an interface to it, which is of no use to anyone, etc.